src/ol_infrastructure/components/services/k8s.py

```diff
@@ -0,0 +1,311 @@
+                                config_map=kubernetes.core.v1.ConfigMapVolumeSourceArgs(
+                                    name=application_nginx_configmap.metadata.name,
+                                    items=[
+                                        kubernetes.core.v1.KeyToPathArgs(
+                                            key="web.conf",
+                                            path="web.conf",
+                                        ),
+                                    ],
+                                ),
+                            ),
+                        ],
+                        init_containers=init_containers,
+                        dns_policy="ClusterFirst",
+                        containers=[
+                            # nginx container infront of uwsgi
+                            kubernetes.core.v1.ContainerArgs(
+                                name="nginx",
+                                image="nginx:1.9.5",
+                                ports=[
+                                    kubernetes.core.v1.ContainerPortArgs(
+                                        container_port=DEFAULT_NGINX_PORT
+                                    )
+                                ],
+                                image_pull_policy="IfNotPresent",
+                                resources=kubernetes.core.v1.ResourceRequirementsArgs(
+                                    requests={"cpu": "50m", "memory": "64Mi"},
+                                    limits={"cpu": "100m", "memory": "128Mi"},
+                                ),
+                                volume_mounts=[
+                                    kubernetes.core.v1.VolumeMountArgs(
+                                        name="staticfiles",
+                                        mount_path="/src/staticfiles",
+                                    ),
+                                    kubernetes.core.v1.VolumeMountArgs(
+                                        name="nginx-config",
+                                        mount_path="/etc/nginx/conf.d/web.conf",
+                                        sub_path="web.conf",
+                                        read_only=True,
+                                    ),
+                                ],
+                            ),
+                            # Actual application run with uwsgi
+                            kubernetes.core.v1.ContainerArgs(
+                                name="application-app",
+                                image=f"mitodl/unified-application-app-main:{self.ol_app_k8s_config.application_docker_tag}",
+                                ports=[
+                                    kubernetes.core.v1.ContainerPortArgs(
+                                        container_port=DEFAULT_UWSGI_PORT
+                                    )
+                                ],
+                                image_pull_policy="IfNotPresent",
```

Comment:
We will want to set the policy to `"Always"` for CI/QA environments to allow for restarting RC deployments.

Reply:
I'd argue we shouldn't have to because they shouldn't be reusing tags / versions but that is a people problem and we haven't addressed it yet.

Reply:
Calling this a no-op as far as the next rev is concerned. @blarghmatey yell if you feel strongly and I'll change it.